storage/innobase/trx/trx0purge.cc

```diff
@@ -1031,10 +1029,26 @@ trx_purge_attach_undo_recs(
 
 		/* Fetch the next record, and advance the purge_sys->iter. */
 		purge_rec->undo_rec = trx_purge_fetch_next_rec(
-			&purge_rec->roll_ptr, &n_pages_handled, node->heap);
+			&purge_rec->roll_ptr, &n_pages_handled, heap);
 
 		if (purge_rec->undo_rec != NULL) {
 
+			table_id_t table_id =
+				trx_undo_rec_get_table_id(purge_rec->undo_rec);
+
+			auto it = table_thr.find(table_id);
+			if (it == table_thr.end()) {
+				thr = run_thrs[free_thr++ % n_purge_threads];
+				table_thr.emplace(table_id, thr);
+			} else {
+				thr = it->second;
+			}
+
+			ut_a(thr != NULL && !thr->is_active);
+			/* Get the purge node. */
+			node = (purge_node_t*) thr->child;
+			ut_a(que_node_get_type(node) == QUE_NODE_PURGE);
+
```

Comment:
I didn't get any response. Almost all the innodb stress tests cover the purge code path.